moto/greengrass/models.py

```diff
@@ -0,0 +1,100 @@
+    def __init__(self, region_name, name):
+        self.region_name = region_name
+        self.name = name
+        self.id = str(uuid.uuid4())
+        self.arn = f"arn:aws:greengrass:{region_name}:{get_account_id()}:greengrass/definition/cores/{self.id}"
+        self.created_at_datetime = datetime.utcnow()
+        self.latest_version = ""
+        self.latest_version_arn = ""
+
+    def to_dict(self):
+        return {
+            "Arn": self.arn,
+            "CreationTimestamp": iso_8601_datetime_with_milliseconds(
+                self.created_at_datetime
+            ),
+            "Id": self.id,
+            "LastUpdatedTimestamp": iso_8601_datetime_with_milliseconds(
+                self.created_at_datetime
+            ),
+            "LatestVersion": self.latest_version,
+            "LatestVersionArn": self.latest_version_arn,
+            "Name": self.name,
+        }
+
+
+class FakeCoreDefinitionVersion(BaseModel):
+    def __init__(self, region_name, core_definition_id, definition):
+        self.region_name = region_name
+        self.core_definition_id = core_definition_id
+        self.definition = definition
+        self.version = str(uuid.uuid4())
+        self.arn = f"arn:aws:greengrass:{region_name}:{get_account_id()}:greengrass/definition/cores/{self.core_definition_id}/versions/{self.version}"
+        self.created_at_datetime = datetime.utcnow()
+
+    def to_dict(self):
+        return {
+            "Arn": self.arn,
+            "CreationTimestamp": iso_8601_datetime_with_milliseconds(
+                self.created_at_datetime
+            ),
+            "Id": self.core_definition_id,
+            "Version": self.version,
+        }
+
+
+class GreengrassBackend(BaseBackend):
+    def __init__(self, region_name=None):
+        super().__init__()
+        self.region_name = region_name
+        self.groups = OrderedDict()
+        self.group_versions = OrderedDict()
+        self.core_definitions = OrderedDict()
+        self.core_definition_versions = OrderedDict()
+        self.device_definitions = OrderedDict()
+        self.device_definition_versions = OrderedDict()
+        self.function_definitions = OrderedDict()
+        self.function_definition_versions = OrderedDict()
+        self.resource_definitions = OrderedDict()
+        self.resource_definition_versions = OrderedDict()
+        self.subscription_definitions = OrderedDict()
+        self.subscription_definition_versions = OrderedDict()
+        self.deployments = OrderedDict()
+
+    def create_core_definition(self, name, initial_version):
+
+        core_definition = FakeCoreDefinition(self.region_name, name)
+        self.core_definitions[core_definition.id] = core_definition
+        self.create_core_definition_version(
+            core_definition.id, initial_version["Cores"]
+        )
+        return core_definition
+
+    def create_core_definition_version(self, core_definition_id, cores):
+
+        definition = {"Cores": cores}
```

Comment:
If we want to check strictly, we need to check IoT thing and certificate exist in the IoT backend.
However, other modules like awslambda are skipping similar checks, so I'm skipping checks here as well.